node/service/src/lib.rs

```diff
@@ -777,7 +792,16 @@ pub fn new_dev(
 		);
 	}
 
-	let spawned_requesters = rpc::spawn_tasks(
+	rpc::spawn_essential_tasks(rpc::SpawnTasksParams {
+		task_manager: &task_manager,
+		client: client.clone(),
+		substrate_backend: backend.clone(),
+		frontier_backend: frontier_backend.clone(),
+		pending_transactions: pending_transactions.clone(),
+		filter_pool: filter_pool.clone(),
+	});
+
+	let tracing_requesters = rpc::tracing::spawn_tracing_tasks(
```

Comment:
Adds #[cfg(feature = "evm-tracing")] on top of this statement. And in the other case create an empty RpcRequester by hand.